pkg/destroy/openstack/openstack.go

```diff
@@ -422,6 +452,28 @@ func deletePorts(opts *clientconfig.ClientOpts, listOpts ports.ListOpts, logger
 				}
 			}
 
+			// If there is a security group created by cloud-provider-openstack we should find it and delete it.
+			// We'll look through the ones on each of the ports and attempt to remove it from the port and delete it.
+			// Most of the time it's a conflict, but last port should be guaranteed to allow deletion.
+			// TODO(dulek): Currently this is the only way to do it and if delete fails there's no way to get back to
+			//              that SG. This is bad and we should make groups created by CPO tagged by cluster ID ASAP.
+			ports.Update(conn, port.ID, ports.UpdateOpts{
+				SecurityGroups: &[]string{}, // We can just detach all, we're deleting this port anyway.
+			})
+			for _, groupID := range port.SecurityGroups {
```

Comment:
It's a bit confusing to use `port.SecurityGroups` after [resetting the port SGs](https://github.com/openshift/installer/pull/7378/files#diff-e480f6c0393b6788fe8352f4f8bdc63f021f9e84de632a21607aab8b414c6e3fR460). I suggest saving the SGs list into a variable.